src/main/kotlin/org/jetbrains/plugins/bsp/server/tasks/CollectProjectDetailsTask.kt

```diff
@@ -155,6 +155,34 @@ public class CollectProjectDetailsTask(project: Project, private val taskId: Any
 
   private fun calculateAllUniqueJdkInfos(projectDetails: ProjectDetails): Set<JvmBuildTarget> = projectDetails.targets.mapNotNull(::extractJvmBuildTarget).toSet()
 
+  private fun calculateAllPythonSdkInfosSubtask(projectDetails: ProjectDetails?) {
+    projectDetails?.let {
+      bspSyncConsole.startSubtask(taskId, "calculate-all-python-sdk-infos", "Calculating all python sdk infos...")
+      pythonSdks = logPerformance("calculate-all-python-sdk-infos") { calculateAllPythonSdkInfos(projectDetails) }
+      bspSyncConsole.finishSubtask("calculate-all-python-sdk-infos", "Calculating all python sdk infos done!")
+    }
+  }
+
+  private data class PythonSdk (
+    val name: String,
+    val interpreter: String,
+    val dependencies: List<DependencySourcesItem>
+  )
+
+  private fun createPythonSdk(target: BuildTarget, dependenciesSources: List<DependencySourcesItem>): PythonSdk? {
+    val pythonInfo = extractPythonBuildTarget(target) ?: return null
+
+    return PythonSdk(
+      "${target.id.uri}-${pythonInfo.version}",
+      pythonInfo.interpreter,
+      dependenciesSources
+    )
+  }
+
+  private fun calculateAllPythonSdkInfos(projectDetails: ProjectDetails): Set<PythonSdk> {
+    return projectDetails.targets.mapNotNull { createPythonSdk(it, projectDetails.dependenciesSources) }.toSet()
```

Comment:
u need to filter dependences for the current target - now u r passing all the deps
something like: `projectDetails.dependenciesSources.filter { a -> a.target.uri == it.id.uri}` but in a nicer way pls